smoke/LeafDevice/details/Details.cs

```diff
@@ -49,6 +49,7 @@ public class Details
             string trustedCACertificateFileName,
             string edgeHostName,
             string edgeDeviceId,
+            string protocol,
```

Comment:
If we are adding protocol, we can remove useWebSockets. Protocol values can be - 
Mqtt, MqttWs, Amqp and AmqpWs.. just like for the product.